examples/vision/siamese.py

```diff
@@ -0,0 +1,427 @@
+"""
+Title: Siamese network with a contrastive loss
+Author: Mehdi
+Date created: 2021/05/06
+Last modified: 2021/05/06
+Description: Similarity learning using siamese network with contrastive loss.
+"""
+
+"""
+## Introduction
+
+[Siamese Network](https://en.wikipedia.org/wiki/Siamese_neural_network)
+is any Neural Network which share weights between two or more sister networks,
+each producing embedding vector of its respective input and these embeddings
+are then passed through some
+[distance heuristic](https://developers.google.com/machine-learning/clustering/similarity/measuring-similarity)
+to find the distance between them. This distance is later used to increase the
+contrast between embeddings of inputs of different classes and decrease it with
+that of similar class by employing some loss function, with the main objective
+of contrasting [vector spaces](https://en.wikipedia.org/wiki/Vector_space)
+from which these sample inputs were taken.
+"""
+
+"""
+## Setup
+"""
+
+import random
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+import matplotlib.pyplot as plt
+
+"""
+Define some hyperparameters
+"""
+
+epochs = 10
+batch_size = 16
+# margin for constrastive loss
+margin = 1
+
+"""
+## Load the MNIST dataset
+"""
+(x_train_val, y_train_val), (x_test, y_test) = keras.datasets.mnist.load_data()
+
+# Change the data type to a floating point format
+x_train_val = x_train_val.astype("float32")
+x_test = x_test.astype("float32")
+
+
+"""
+Use list slicing to split train_val data into `train` and `val`
+"""
+
+# Keep 50% of train_val  in validation set
+x_train, x_val = x_train_val[0:30000], x_train_val[30000:]
+y_train, y_val = y_train_val[0:30000], y_train_val[30000:]
+del x_train_val
+del y_train_val
+
```

Comment:
`0` is implicit in `x_train_val[0:30000]`; it is more pythonic as `x_train_val[:30000]`.

Additionally, you can simplify `del` to a single line: `
del x_train_val, y_train_val`